App/Syncthing/syncthing_sync_check.sh

```diff
@@ -0,0 +1,284 @@
+        return 0
+    else
+        return 1
+    fi
+}
+
+get_folder_status() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/db/status?folder=$1"
+}
+
+get_device_name() {
+    local device_id="$1"
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/config/devices" | jq -r ".[] | select(.deviceID == \"$device_id\") | .name"
+}
+
+calculate_folder_completion() {
+    local folder_id="$1"
+    local folder_status=$(get_folder_status "$folder_id")
+    local global_bytes=$(echo "$folder_status" | jq '.globalBytes // 0')
+    local need_bytes=$(echo "$folder_status" | jq '.needBytes // 0')
+
+    if [ "$global_bytes" -eq 0 ]; then
+        echo "100"
+    else
+        local completed_bytes=$((global_bytes - need_bytes))
+        local percentage=$((completed_bytes * 100 / global_bytes))
+        echo $percentage
+    fi
+}
+
+get_completion() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/db/completion?device=$1&folder=$2"
+}
+
+calculate_total_completion() {
+    local device="$1"
+    local folder_id="$2"
+    local completion=$(get_completion "$device" "$folder_id")
+
+    local global_bytes=$(echo "$completion" | jq '.globalBytes // 0')
+    local need_bytes=$(echo "$completion" | jq '.needBytes // 0')
+    local need_deletes=$(echo "$completion" | jq '.needDeletes // 0')
+    local need_items=$(echo "$completion" | jq '.needItems // 0')
+
+    if [ "$need_deletes" -gt 0 ] || [ "$need_items" -gt 0 ] || [ "$need_bytes" -gt 0 ]; then
+        local percentage=$((100 * (global_bytes - need_bytes) / global_bytes))
+        echo $percentage
+    else
+        echo "100"
+    fi
+}
+
+monitor_sync_status() {
+    local mode="$1"
+    local folders=$(get_folders)
+    local devices=$(get_devices)
+
+    log_message "Monitoring sync status in $mode mode"
+    display -t "Syncthing Check:
+
+Press START to cancel"
+
+    if [ "$mode" = "shutdown" ]; then
+        force_rescan
+        sleep 1
+    elif [ "$mode" = "startup" ]; then
+        force_rediscovery
+        # THIS MIGHT NOT BE NEEDED, WE MIGHT BE ABLE TO JUST SLEEP 1
+        # sleep 1
+        sleep 5
+    fi
+
+    # Check if any devices are online
+    if ! are_devices_online; then
+        log_message "No devices are online. Exiting sync check."
+        display -t "No devices online"
+        sleep 1
+        return 1
+    fi
+
+    while true; do
+        local all_synced=true
+        local summary=""
+        local status_lines=""
+
+        # Check if the Start button was pressed
+        if tail -n1 "/var/log/messages" | grep -q "enter_pressed 0"; then
+            log_message "Sync check cancelled by user"
+            return 1
+        fi
+
+        # Remove the status files at the beginning of each iteration
+        rm -f /tmp/sync_status
+        rm -f /tmp/sync_display.txt
+    
+        # TODO: Consider refactoring to use in-memory approach instead of temporary files
+        # for performance improvement and reduced I/O. Current file-based method:
```

Comment:
Is this spending a considerable amount of time on file io? I'd be surprised if the performance was meaningfully different between a file and memory unless you're reading or writing to it a lot.

Reply:
You're right, i think I was exaggerating here on potential resource usage. Ill remove this TODO:

f5fdab1d534e98d6956c40e79a7fb85695a4d390